docs/macros.py

```diff
@@ -0,0 +1,77 @@
+"""Macros and filters made available in Markdown pages."""
+
+import asyncio
+from itertools import chain
+from pathlib import Path
+
+import httpx
+import toml
+from jinja2 import StrictUndefined
+from jinja2.sandbox import SandboxedEnvironment
+from pip._internal.commands.show import search_packages_info  # noqa: WPS436 (no other way?)
+
+
+def get_credits_data() -> dict:
```

Comment:
I was thinking, it sucks that we'll have to ping the pypi site on every build. But no easy solution, so whatever.

However, one easy thing to do is `@functools.lru_cache()`, that'll at least make it stick within repeats in `mkdocs serve`

Reply:
Yeah I was thinking about caching results on the disk. It's really important if the cache is out-of-date.